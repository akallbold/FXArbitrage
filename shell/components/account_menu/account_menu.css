@import '../../../../node_modules/playa-styleguide/src/stylesheets/variables.css';

.accountIconContainer {
  position: relative;
  outline: 0;
  line-height: 30px;
}

.accountIcon {
  height: 20px;
  vertical-align: middle;
  fill: var(--white);
  cursor: pointer;
}

.modal {
  top: 38px;
  right: calc(50% - 12px);
  width: 320px;
  position: absolute;
  display: block;
  background-color: var(--white);
  border: 1px solid var(--grey);
  border-radius: 2px;
  box-shadow: 0px 10px 50px -20px var(--text-grey);
  padding: 20px;
  padding-bottom:0;
  z-index: var(--z-index-9);
}

.modal.isHidden {
  display: none;
}

.modal:before {
  content: '';
  top: -5px;
  right: 7px;
  height: 8px;
  position: absolute;
  width: 8px;
  transform: rotate(45deg);
  background-color: var(--white);
  border-left: 1px solid var(--grey);
  border-top: 1px solid var(--grey);
}

.profileImg {
  float: left;
  height: 70px;
  width: 70px;
  margin-right: 20px;
}

.userDetails {
  margin-bottom: 20px;
}

.name, .email {
  margin-top: 0;
  margin-bottom: .5em;
}

.email {
  color: var(--text-grey-light);
}

.firstLinkMargin {
  margin-top:50px;
}

.link {
  padding: 8.5px 0;
  border-top: 1px solid var(--grey);
}
