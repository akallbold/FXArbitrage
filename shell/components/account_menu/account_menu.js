import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { get } from 'lodash'
import {
  viewProfile,
  signOut,
  redirectToHelp,
  hideAccountMenu,
  showAccountMenu
} from 'actions/account_menu_actions'
import { getUser } from 'selectors/user_selectors'
import { isAccountMenuHidden } from 'selectors/account_menu_selectors'
import { Button, AccountIcon } from 'playa-styleguide'
import Styles from './account_menu.css'

export const AccountMenu = ({
  isHidden,
  user,
  ...restProps
}) => {
  const userAvatarURL = get(user, 'avatar.thumb_url', '')

  const onBlur = (event) => {
    if (!event.currentTarget.contains(event.relatedTarget)) {
      restProps.hideAccountMenu()
    }
  }

  return (
    <div
      className={Styles.accountIconContainer}
      tabIndex='0'
      onBlur={onBlur}
    >
      <AccountIcon
        onClick={isHidden ? restProps.showAccountMenu : restProps.hideAccountMenu}
        className={Styles.accountIcon}
      />
      <div className={`${Styles.modal} ${isHidden ? Styles.isHidden : ''}`}>
        <div className={Styles.userDetails}>
          <img className={Styles.profileImg} src={userAvatarURL} alt='profile' />
          <p className={Styles.name}>{user.firstName} {user.lastName}</p>
          <p className={Styles.email}>{user.email}</p>
          <Button
            priority='secondary'
            size='fitToText'
            onClick={restProps.viewProfile}
          >View Profile
          </Button>
        </div>
        <div className={`${Styles.link} ${Styles.firstLinkMargin}`}>
          <Button
            priority='secondaryText'
            size='small'
            onClick={restProps.redirectToHelp}
          >Help
          </Button>
        </div>
        <div className={Styles.link}>
          <Button
            priority='secondaryText'
            size='small'
            onClick={restProps.signOut}
          >Sign Out
          </Button>
        </div>
      </div>
    </div>
  )
}

AccountMenu.propTypes = {
  // mapStateToProps
  isHidden: PropTypes.bool,
  user: PropTypes.object,
  // mapDispatchToProps
  showAccountMenu: PropTypes.func,
  hideAccountMenu: PropTypes.func,
  viewProfile: PropTypes.func,
  redirectToHelp: PropTypes.func,
  signOut: PropTypes.func
}

AccountMenu.defaultProps = {
  isHidden: true,
  user: {},
  showAccountMenu: null,
  hideAccountMenu: null,
  viewProfile: null,
  redirectToHelp: null,
  signOut: null
}

const mapStateToProps = state => ({
  user: getUser(state) || {},
  isHidden: isAccountMenuHidden(state)
})

const mapDispatchToProps = {
  viewProfile,
  signOut,
  redirectToHelp,
  hideAccountMenu,
  showAccountMenu
}

const AccountMenuConnected = connect(
  mapStateToProps,
  mapDispatchToProps
)(AccountMenu)

export default AccountMenuConnected
