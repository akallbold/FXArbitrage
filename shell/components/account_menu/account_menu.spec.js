import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AccountMenu } from './account_menu'

const user = {
  firstName: 'Horse',
  lastName: 'Badorties',
  email: 'horse.badorties@thefanman.com'
}

const func = () => {}

describe('<AccountMenu />', () => {
  it('should render given no props', () => {
    const component = shallow(
      <AccountMenu />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render given all props', () => {
    const component = shallow(
      <AccountMenu
        isHidden
        user={user}
        showAccountMenu={func}
        hideAccountMenu={func}
        viewProfile={func}
        redirectToHelp={func}
        signOut={func}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
