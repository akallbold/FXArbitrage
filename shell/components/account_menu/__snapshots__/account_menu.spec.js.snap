// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountMenu /> should render given all props 1`] = `
<div
  className="accountIconContainer"
  onBlur={[Function]}
  tabIndex="0"
>
  <test-file-stub
    className="accountIcon"
    onClick={[Function]}
  />
  <div
    className="modal isHidden"
  >
    <div
      className="userDetails"
    >
      <img
        alt="profile"
        className="profileImg"
        src=""
      />
      <p
        className="name"
      >
        Horse
         
        Badorties
      </p>
      <p
        className="email"
      >
        horse.badorties@thefanman.com
      </p>
      <Button
        className=""
        component={null}
        disabled={false}
        href=""
        onClick={[Function]}
        priority="secondary"
        size="fitToText"
      >
        View Profile
      </Button>
    </div>
    <div
      className="link firstLinkMargin"
    >
      <Button
        className=""
        component={null}
        disabled={false}
        href=""
        onClick={[Function]}
        priority="secondaryText"
        size="small"
      >
        Help
      </Button>
    </div>
    <div
      className="link"
    >
      <Button
        className=""
        component={null}
        disabled={false}
        href=""
        onClick={[Function]}
        priority="secondaryText"
        size="small"
      >
        Sign Out
      </Button>
    </div>
  </div>
</div>
`;

exports[`<AccountMenu /> should render given no props 1`] = `
<div
  className="accountIconContainer"
  onBlur={[Function]}
  tabIndex="0"
>
  <test-file-stub
    className="accountIcon"
    onClick={null}
  />
  <div
    className="modal isHidden"
  >
    <div
      className="userDetails"
    >
      <img
        alt="profile"
        className="profileImg"
        src=""
      />
      <p
        className="name"
      >
         
      </p>
      <p
        className="email"
      />
      <Button
        className=""
        component={null}
        disabled={false}
        href=""
        onClick={null}
        priority="secondary"
        size="fitToText"
      >
        View Profile
      </Button>
    </div>
    <div
      className="link firstLinkMargin"
    >
      <Button
        className=""
        component={null}
        disabled={false}
        href=""
        onClick={null}
        priority="secondaryText"
        size="small"
      >
        Help
      </Button>
    </div>
    <div
      className="link"
    >
      <Button
        className=""
        component={null}
        disabled={false}
        href=""
        onClick={null}
        priority="secondaryText"
        size="small"
      >
        Sign Out
      </Button>
    </div>
  </div>
</div>
`;
